app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,70 @@
+class FoodsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_food, only: %i[show edit update destroy]
+
+  # GET /foods or /foods.json
+  def index
+    @foods = Food.where(user: current_user)
```

Comment:
Please solve the `N+1` problem in your controller index action. A very common solution is eager loading. You may learn more about it [here](https://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations).